types/scenegraph.d.ts

```diff
@@ -10,6 +10,122 @@ declare interface ScaleFactor {
     scaleY: number;
 }
 
+/**
+ * (**Since:** XD 29)
+ *
+ * Stores metadata accessible to multiple plugins, separated into silos by plugin ID. Your plugin can read & write the storage for its own plugin ID, but storage for other plugin IDs is *read-only*.
+ *
+ * Each per-plugin storage silo is a collection of key-value pairs. Keys and values must both be strings.
+ *
+ * *Each* scenenode has its own metadata storage, accessed via `SceneNode.sharedPluginData`. To store general metadata that is not specific to one scenenode, use `sharedPluginData` on the document's scenegraph root.
+ * @example ```js
+ * // This example shows how to save & retrieve rich JSON data in shared metadata storage.
+ // See below for simpler examples of using individual APIs.
+ const PLUGIN_ID = "<your manifest's plugin ID here>";
+ let richObject = {
+    list: [2, 4, 6],
+    name: "Hello world"
+ };
+ node.sharedPluginData.setItem(PLUGIN_ID, "richData", JSON.stringify(richObject));
+
+ // Later on...
+ // (This could be in a different plugin, if it passes the original plugin's ID here)
+ let jsonString = node.sharedPluginData.getItem(PLUGIN_ID, "richData");
+ if (jsonString) {  // may be undefined
+    let richObjectCopy = JSON.parse(jsonString);
+    console.log(richObjectCopy.list.length);  // 3
+ }
```

Comment:
The example markup is totally broken in VSCode (doesn't render correctly). Putting the triple-tick onto its own line above/below the code block and adding stars to each line in the comment appears to have fixed the issue. See:

```suggestion
 * @example 
 * ```
 * // This example shows how to save & retrieve rich JSON data in shared metadata storage.
 * // See below for simpler examples of using individual APIs.
 *  const PLUGIN_ID = "<your manifest's plugin ID here>";
 *  let richObject = {
 *    list: [2, 4, 6],
 *    name: "Hello world"
 *  };
 *  node.sharedPluginData.setItem(PLUGIN_ID, "richData", JSON.stringify(richObject));
 *
 * // Later on...
 * // (This could be in a different plugin, if it passes the original plugin's ID here)
 * let jsonString = node.sharedPluginData.getItem(PLUGIN_ID, "richData");
 * if (jsonString) {  // may be undefined
 *   let richObjectCopy = JSON.parse(jsonString);
 *   console.log(richObjectCopy.list.length);  // 3
 * }
 * ```
```